### PR TITLE
Fix get_changed_files in infra/ci.

### PR DESCRIPTION
**Pull request by inferno-chromium**
Fixes https://github.com/google/oss-fuzz/issues/5022

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Could not share code since these are standalone file, also there are some differences around committed and uncommitted changes.

File: infra/ci/build.py [hunk 1]
```diff
@@ -35,16 +35,21 @@
...
 
-def get_changed_files():
+def get_changed_files_output():
   """Returns the output of a git command that discovers changed files."""
-  return subprocess.check_output(['git', 'diff', '--name-only',
-                                  'FETCH_HEAD']).decode()
+  main_branch = subprocess.check_output(
+      ['git', 'rev-parse', '--abbrev-ref', 'origin/HEAD']).strip().decode()
+  branch_commit_hash = subprocess.check_output(
+      ['git', 'merge-base', 'FETCH_HEAD', main_branch]).strip().decode()
```
Comment (oliverchang) [inline infra/ci/build.py:43]: can we just specify origin/HEAD or do we actually need the rev-parse above?

**Review by oliverchang — APPROVED** [review]

File: infra/ci/build.py (hunk 1 above)
Comment (inferno-chromium) [reply infra/ci/build.py:43]: Actually no, so removing in https://github.com/google/oss-fuzz/pull/5056

**Review by jonathanmetzman — COMMENTED** [review]
LGTM

**Pull request merged**